File: os/OS exam/fcfs.cpp
#include<bits/stdc++.h>
using namespace std;
struct process{
    int id,arrival,burst,completion;
}p[100];
int n;
void sort(){
    for(int i=0;i<n-1;i++){
        for(int j=i+1;j<n;j++){
            if(p[i].arrival>p[j].arrival)
            {
                swap(p[i],p[j]);
            }
        }
    }
}
void fcfs(){
    sort();
    float averageturn=0,averagewait=0;
    int time=0;
    for(int i=0;i<n;i++){
        p[i].completion=p[i].burst+time;
        time=p[i].completion;
    }
    cout<<"\nprocessNo Arrival Burst Completion Turn Wait"<<endl;
    for(int i=0;i<n;i++){
        cout<<"\t"<<p[i].id<<"\t"<<p[i].arrival<<"\t"<<p[i].burst<<"\t"<<p[i].completion<<"\t"<<p[i].completion-p[i].arrival<<"\t"<<p[i].completion-p[i].arrival-p[i].burst<<endl;
        averageturn+=p[i].completion-p[i].arrival;
        averagewait+=p[i].completion-p[i].arrival-p[i].burst;
    }
    cout<<"Turn around time:"<<averageturn/n<<endl<<"waiting time:"<<averagewait/n<<endl;
}
int main(){
 cout<<"enter total no of process:";
 cin>>n;
 for(int i=0;i<n;i++){
    cout<<"enter the arrival time for the process :"<<(i+1);
    cin>>p[i].arrival;
    p[i].id=i+1;
 }
  for(int i=0;i<n;i++){
    cout<<"enter the burst time for the process :"<<(i+1);
    cin>>p[i].burst;
 }
fcfs();
}

File: os/OS exam/fcfs.java
import java.util.*; 

public class fcfs {
    static class process{
        int id,arrival,burst,completion;
        process(int id,int arrival, int burst){
            this.id=id;
            this.arrival=arrival;
            this.burst=burst;
        }
    }
    static process p[];
    static int n;

    static void readArray(process array[]){
        Scanner sc=new Scanner(System.in);
        System.out.println("for arrival:");
        for(int i=0;i<n;i++){
            System.out.println("element "+(i+1)+" :");
            array[i]=new process(i+1, sc.nextInt(), 0);
         }
         System.out.println("for burst:");
         for(int i=0;i<n;i++){
            System.out.println("element "+(i+1)+" :");
            array[i].burst=sc.nextInt();    
        }
        sc.close();
    }
    static void sort(process array[]){
        for(int i=0;i<n-1;i++){
            for(int j=i+1;j<n;j++){
                if(array[i].arrival>array[j].arrival){
                    process temp=array[i];
                    array[i]=array[j];
                    array[j]=temp;
                }
            }
        }
    }
    static void firstComeFirstServe(process array[]){
        sort(array);
        float turn=0;
        float wait=0;
        int time=0;
        for(int i=0;i<n;i++){
            array[i].completion=time+array[i].burst;
            time=array[i].completion;
        }
        System.out.println("Process Arrival Burst Completion Turn Wait");
        for(int i=0;i<n;i++){
            System.out.print(array[i].id+"\t"+array[i].arrival+"\t"+array[i].burst+"\t"+array[i].completion+"\t"+(array[i].completion-array[i].arrival)+"\t"+(array[i].completion-array[i].arrival-array[i].burst)+"\n");
            turn+=array[i].completion-array[i].arrival;
            wait+=array[i].completion-array[i].arrival-array[i].burst;
        }
        System.out.println("average turnaround time="+turn/n);
        System.out.println("average waiting time="+wait/n);

    }
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        System.out.println("enter the number of processes:");
        n=sc.nextInt();
        p=new process[n];
        readArray(p);
        firstComeFirstServe(p);
        sc.close();    
    }
    
}
